Sources/HummingbirdWebSocket/WebSocketInboundStream.swift

```diff
@@ -104,9 +89,49 @@ public final class WebSocketInboundStream: AsyncSequence, Sendable {
 
             return nil
         }
-    }
 
-    public typealias Element = WebSocketDataFrame
+        /// Return next WebSocket messsage, while dealing with any other frames
+        ///
+        /// A WebSocket message can be fragmented across multiple WebSocket frames. This
+        /// function collates fragmented frames until it has a full message
+        public mutating func nextMessage(maxSize: Int) async throws -> WebSocketMessage? {
+            var frameSequence: WebSocketFrameSequence
+            // parse first frame
+            guard let frame = try await self.next() else { return nil }
+            switch frame.opcode {
+            case .text, .binary:
+                frameSequence = .init(frame: frame)
+                if frame.fin {
+                    let collatedFrame = frameSequence.collapsed
+                    if let finalFrame = WebSocketMessage(frame: collatedFrame) {
```

Comment:
I mean here we are already certain we have a non-nil message. Otherwise our entire logic would be wrong. But sure, we can do the `if let` too 🤷‍♂️

Reply:
Yep you're right there. It was something I considered but forgot about.

Reply:
I moved the conversion into WebSocketFrameSequence and added a `!` and an `assert`. Code looks cleaner as well.